example-dotnet/Program.cs

```diff
@@ -11,8 +22,134 @@ static void Main(string[] args)
             Say.hello("example");
             Console.WriteLine("Hello World!");
 
+	    // Usage of BCrypt.Net API
+
             var hashedPassword = BCrypt.Net.BCrypt.HashPassword("my-secret-password");
             BCrypt.Net.BCrypt.Verify("wrong-password", hashedPassword);
+
+            // Usage of SharpZipLib API
+
+	    // This can trigger an exploit, see:
+	    // https://github.com/icsharpcode/SharpZipLib/issues/232
+	    // The usage of SharpZipLib API here will be reported as
+	    // may be reaching one of the methods affected by an
+	    // update from vulnerable version to the safe version.
+
+	    // `FastZip().ExtractZip()` from SharpZipLib 1.0.0-alpha1
+	    // is known to be vulnerable (CVE-2018-1002208), see also:
+	    // https://github.com/icsharpcode/SharpZipLib/issues/232. Latest
+	    // version 1.1.0 without the vulnerability has a
+	    // compatible API. The software can be updated to
+	    // 1.0.0-rc1 to resolve the issue. 1.0.0-alpha1 includes
+	    // netstandard1.3 framework version, while 1.0.0-rc1
+	    // includes net45 and netstandard2.0 framework
+	    // versions. The usage of the vulnerable API is commented
+	    // out, and we instead opted to include a simple access to
+	    // a constant in the library. This is to exemplify the
+	    // negative case of breaking update over different sets of
+	    // frameworks between the current version and the
+	    // recommended update.
+
+            // Console.WriteLine("Unzipping example ZIP file using SharpZipLib");
+            // new FastZip().ExtractZip(@"resources" + Path.DirectorySeparatorChar + "example.zip", @".", null);

```

Comment:
Thanks for checking the platforms so thoroughly. I don't understand why we explain everything only to comment out the use though; we also don't pick up non-method changes (constants), so there's effectively no use of the library.

Does "negative case of breaking update over different sets of frameworks" just mean there is no breaking change? It's not clear to me what the negation applies to.

Reply:
We should ensure that our "no breaking change" reports appear _despite_ having a use. If there's no use to begin with, we may say "No" trivially and the example wouldn't be meaningful.

Reply:
@dariusf Hopefully with the tables that I have included in the code this is clearer.

There is a no breaking change that appears despite having a use, with the recurly-client-api API usage of a property (calling of a getter).